pytest_mergify/__init__.py

```diff
@@ -85,14 +93,68 @@ def tracer(self) -> typing.Optional[opentelemetry.trace.Tracer]:
         return self.mergify_tracer.tracer
 
     def pytest_sessionstart(self, session: _pytest.main.Session) -> None:
+        self.has_error = False
+
         if self.tracer:
             self.session_span = self.tracer.start_span(
                 "pytest session start",
                 attributes={
                     "test.scope": "session",
                 },
             )
-        self.has_error = False
+
+        if not utils.is_in_ci():
+            return
+
+        if self.mergify_tracer.repo_name is None:
+            self.quarantine_warning_logs.append(
+                "No valid repository name found, skipping CI Insights Quarantine setup"
+            )
+            return
+
+        # Query all the quarantined tests
+        try:
+            owner, repository = self.mergify_tracer.repo_name.split("/")
+        except ValueError:
+            self.quarantine_warning_logs.append(
+                f"Repository name '{self.mergify_tracer.repo_name}' has an unexpected format (expected 'owner/repository'), skipping CI Insights Quarantine setup"
+            )
+            return
+
+        branch_name = utils.get_branch_name_for_quarantine()
+        if branch_name is None:
+            self.quarantine_warning_logs.append(
+                "Could not find a branch name, which is required to use CI Insights Quarantine",
+            )
+            return
+
+        url = f"{self.mergify_tracer.api_url}/v1/ci/{owner}/repositories/{repository}/quarantines"
+
+        try:
+            quarantine_resp: requests.Response = requests.get(
+                url,
+                params={"branch": branch_name},
+                timeout=10,
+            )
+        except requests.ConnectionError as exc:
+            self.quarantine_warning_logs.append(
+                f"Failed to connect to Mergify's API, tests won't be quarantined. Error: {str(exc)}"
+            )
+            return
+
+        if quarantine_resp.status_code == 402:
+            # No CI Insights Quarantine subscription, skip it.
+            return
+
+        try:
+            quarantine_resp.raise_for_status()
+        except requests.HTTPError as exc:
+            self.quarantine_warning_logs.append(
+                f"Error when querying Mergify's API, tests won't be quarantined. Error: {str(exc)}"
+            )
+            return
+
+        self.quarantined_tests = quarantine_resp.json()["quarantined_tests"]
```

Comment:
small suggestions to improve the separation of concerns, we can move all the code we have above to a dedicated module or least a function:
```suggestion
        self.quarantined_tests = get_quarantined_tests(repo_name, branch_name)
```